pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -295,27 +348,49 @@ func ValidateFlowSchemaCondition(condition *flowcontrol.FlowSchemaCondition, fld
 // ValidatePriorityLevelConfiguration validates priority-level-configuration.
 func ValidatePriorityLevelConfiguration(pl *flowcontrol.PriorityLevelConfiguration) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&pl.ObjectMeta, false, ValidatePriorityLevelConfigurationName, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidatePriorityLevelConfigurationSpec(&pl.Spec, pl.Name, field.NewPath("spec"))...)
+	specPath := field.NewPath("spec")
+	switch pl.Name {
+	case flowcontrol.PriorityLevelConfigurationNameExempt, flowcontrol.PriorityLevelConfigurationNameCatchAll:
+		src := bootstrap.DefaultPriorityLevelConfigurationExempt
+		if pl.Name == flowcontrol.PriorityLevelConfigurationNameCatchAll {
+			src = bootstrap.DefaultPriorityLevelConfigurationDefault
+		}
+		var rightSpec flowcontrol.PriorityLevelConfigurationSpec
+		err := legacyscheme.Scheme.Converter().Convert(&src.Spec, &rightSpec, 0, &conversion.Meta{})
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(specPath, pl.Spec, "Conversion failed: "+err.Error()))
+		} else if !apiequality.Semantic.DeepEqual(pl.Spec, rightSpec) {
+			allErrs = append(allErrs, field.Invalid(specPath, pl.Spec, fmt.Sprintf("spec of '%s' must equal the fixed value", pl.Name)))
+		}
+	default:
+		allErrs = append(allErrs, ValidatePriorityLevelConfigurationSpec(&pl.Spec, pl.Name, specPath)...)
+	}
 	allErrs = append(allErrs, ValidatePriorityLevelConfigurationStatus(&pl.Status, field.NewPath("status"))...)
 	return allErrs
 }
 
 // ValidatePriorityLevelConfigurationUpdate validates the update of priority-level-configuration.
 func ValidatePriorityLevelConfigurationUpdate(old, pl *flowcontrol.PriorityLevelConfiguration) field.ErrorList {
+	if (pl.Name == flowcontrol.PriorityLevelConfigurationNameExempt || pl.Name == flowcontrol.PriorityLevelConfigurationNameCatchAll) && !apiequality.Semantic.DeepEqual(old.Spec, pl.Spec) {
```

Comment:
am in favor of moving this to a dedicated admission-controller

Reply:
see previous

Reply:
We can always move it to one in a followup.

The only benefit I see to an admission plugin is that it'd give a place to return an error message if the conversion in init functions fails. (Consistency is another possible benefit but it doesn't motivate me to ask for a change in this case.)

Reply:
@lavalamp: does your reply apply to a different piece of code?  Right here the code is validating an update by comparing old and new without conversion.

Reply:
On further thought, this code is not even correct.  A given apiserver should accept an update that sets to the Spec of a mandatory object to the value that said apiserver insists on, regardless of whether that is the previous value of the Spec (this sort of thing matters in mixed-version scenarios).  I am going to revise accordingly (I think this will simply amount to reverting the change).